syft/pkg/cataloger/dart/parse_pubspec_lock.go

```diff
@@ -31,13 +32,39 @@ type pubspecLockPackage struct {
 }
 
 type pubspecLockDescription struct {
+	isString bool
+	str      string
+	object   pubspecLockDescriptionObject
+}
+
+type pubspecLockDescriptionObject struct {
```

Comment:
I don't think this change in structure is necessary -- this can all be handled in the UnmarshalYAML